### ISSUE TITLE
Consider publishing information about the DockerTools changes in 15.3 preview release notes

### ISSUE THREAD

**Issue by dls314** [Microsoft/DockerTools#32]
I noticed that in 15.3 preview 6 the DockerTools don't seem to create the same file set (the docker-compose-override.vs.debug.yml and docker-compose-override.vs.release.yml files aren't created).

Is there anywhere I can read more information about these changes?
I tried the preview release notes, but didn't see the information there.

Thanks!

**Comment by pbering** [comment]
Even worse, when updating VS and the docker project has been upgrade to v2.0, the release and debug files are deleted! No explanation on what to do then? I was using those files... Can't even add them again, then I get an error:

`1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\MSBuild\Sdks\Microsoft.Docker.Sdk\build\Microsoft.VisualStudio.Docker.Compose.targets(279,5): error : Operation is not valid due to the current state of the object.
`

What is the intention with this? How can we differentiate our compose files per build configuration?

**Comment by RehanSaeed** [comment]
I'd like information about how to use the `docker-compose.ci.build.yml` file on my build agent. I'm not sure what magic incantation I need to chant to get docker to build my image with a particular set of tags (`latest` and a second tag containing a version number`). Some documentation would be nice. That said, this project moves the cheese so often, I'm not sure how long the documentation would still be valid.

**Comment by alexzaytsev-newsroomly** [comment]
I second that, particularly in case of VSTS, some documentation on a complete workflow - docker-compose project on dev machine and how to build the same on CI would be really helpful.

**Comment by dmarlow** [comment]
@pbering did you ever end up finding a fix to your problem? I'm running into the same thing:

```
1>  C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.Docker.Sdk\build\Microsoft.VisualStudio.Docker.Compose.targets(281,5): error : Operation is not valid due to the current state of the object.
1>Done building target "DockerCreateComposeVsGeneratedFiles" in project "docker-compose.dcproj" -- FAILED.
```

**Comment by pbering** [comment]
@dmarlow not exactly... ended up just stopping using those files. Later on in 15.4 the use of a custom entrypoint was also broken so can't use that anymore either (the tooling overwrites it)

**Comment by alexvy86** [comment]
I've been running into the same issue as @pbering when trying to build the `docker-compose` project and I managed to figure out what was causing it in my case.

For clarity, here's the error again, as seen in the Build output in Visual Studio:

```
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\MSBuild\Sdks\Microsoft.Docker.Sdk\build\Microsoft.VisualStudio.Docker.Compose.targets(297,5): error : Operation is not valid due to the current state of the object.
1>Done building project "docker-compose.dcproj" -- FAILED.`
```

For certain scenarios, I wanted to use `docker-compose` just to start a bunch of containers from images that are already built (instead of creating an empty container and then mount a directory with my local code, for debugging purposes). For those scenarios I removed the `build` section of every service in the docker-compose file, and this seems to be what causes the error. As soon as I add a `build` section to at least one service in the docker-compose file (in either `docker-compose.yml` or `docker-compose.override.yml`), the project build succeeds.
